src/main/java/io/github/opencubicchunks/cubicchunks/world/level/chunk/LevelCube.java

```diff
@@ -1090,29 +1101,11 @@ public void unpackTicks() {
     }
 
     public void postLoad() {
+
         if (this.postLoad != null) {
             this.postLoad.accept(this);
             this.postLoad = null;
         }
-        // TODO heightmap stuff should probably be elsewhere rather than here.
-        ChunkPos pos = this.cubePos.asChunkPos();
-        HeightmapStorage storage = ((CubicServerLevel) this.level).getHeightmapStorage();
-        for (int x = 0; x < CubeAccess.DIAMETER_IN_SECTIONS; x++) {
-            for (int z = 0; z < CubeAccess.DIAMETER_IN_SECTIONS; z++) {
-
-                // This force-loads the column, but it shouldn't matter if column-cube load order is working properly.
-                LevelChunk chunk = this.level.getChunk(pos.x + x, pos.z + z);
-                ((ColumnCubeMapGetter) chunk).getCubeMap().markLoaded(this.cubePos.getY());
-                for (Map.Entry<Heightmap.Types, Heightmap> entry : chunk.getHeightmaps()) {
-                    Heightmap heightmap = entry.getValue();
-                    SurfaceTrackerWrapper tracker = (SurfaceTrackerWrapper) heightmap;
-                    tracker.loadCube(storage, this);
-                }
-
-                // TODO probably don't want to do this if the cube was already loaded as a CubePrimer
-                ((LightHeightmapGetter) chunk).getServerLightHeightmap().loadCube(storage, this);
-            }
-        }
```

Comment:
This logic was partly duplicated between the promotion constructor and postLoad. I moved all of it into the promotion constructor for now.